### PR TITLE
Centralizing name/rank/etc stuff used in messages for #442

### PR DESCRIPTION
**Pull request by msouth**
There were computations of name/rank/etc all over
the place.  The goal of this is to put it in fewer
places and call from there.

(n.b. I'm not too familiar with the JS object model yet,
this might could use some further refinement.)

### CODE REVIEW

**Comment by ingmargoudt** [comment]
Thanks for the update! Will test it in a couple of hours and return results :)

File: Game/Fighter.js [hunk 1]
```diff
@@ -817,5 +817,23 @@ var Fighter = Actor.extend({
...
         });
+    },
+    NameRankEtc: function() {
+        //guests and users have a positive id, npcs do not
+        var name = (this.id > 0) ? this.name : this.template.prefix + ' ' + this.template.name;
+
+        var rank = 'npc';
+        if (this.id > 0) {
+            rank = this.isGuest ? 'guest' : (this.editor ? 'gm' : 'user');
+            if (unit.chDevNinja == true) rank = 'user';
```
Comment (ingmargoudt) [inline Game/Fighter.js]: instead of unit.chDevNinja, this should be: this.chDevNinja

**Comment by ingmargoudt** [comment]
I was able to log in after the rewrite from unit to this. Logged in as admin and guest and saw purple and orange respectively

File: Game/Fighter.js (hunk 1 above)
Comment (bsparks) [inline Game/Fighter.js]: also please wrap in curlies, no lnline if statements

**Comment by bsparks** [comment]
looks good, approved

**Comment by ingmargoudt** [comment]
approved! :)

**Pull request merged**